### DISCUSSION TITLE
WebStorm incompatibility with TypeScript 5.x.x

### DISCUSSION THREAD

**Discussion by arturabg** [pixijs/pixijs#9290] [General]
<img width="408" alt="Screen Shot 2023-03-23 at 22 38 51" src="https://user-images.githubusercontent.com/83285441/227315269-4dcd7cb3-a215-482e-851f-0f73cefd9cd6.png">
<img width="255" alt="Screen Shot 2023-03-23 at 22 39 28" src="https://user-images.githubusercontent.com/83285441/227315460-6eedce64-fc88-4310-a8b3-3ad7e7f9fb42.png">
<img width="392" alt="Screen Shot 2023-03-23 at 22 44 01" src="https://user-images.githubusercontent.com/83285441/227316681-241bf793-2ef0-41e4-af82-ec8bc0d35474.png">

I'm using the last version of Pixi.js and I have some problems
Graphics -> eventMode, on  methods show Unresolved variable eventMode 
Who is can help me ?

**Comment by SuperSodaSea** [comment]
Try:
- Reinstall all PixiJS related package:
```
npm uninstall pixi.js @pixi/events @pixi/graphics @pixi/sound
npm install pixi.js @pixi/events @pixi/graphics @pixi/sound
```
- Or remove `package-lock.json` & `node_modules` and rerun `npm install`

([`@types/pixi.js`](https://www.npmjs.com/package/@types/pixi.js) is not needed now, you can just remove it.)

**Reply by arturabg** [reply]
I tried this but it's still show yet same types error!
Lot of thanks

**Reply by SuperSodaSea** [reply]
Would you like to create a small repo to reproduce your issue?

**Reply by arturabg** [reply]
You mean in my Github dashboard create repo and move my code there?
this is my tsconfig.json file
<img width="405" alt="Screen Shot 2023-03-24 at 13 49 00" src="https://user-images.githubusercontent.com/83285441/227486542-fa0d644a-19a0-4c0a-b7ef-b02d195ccfcc.png">

**Reply by SuperSodaSea** [reply]
Yes, a minimal project that can reproduce your problem.

**Reply by arturabg** [reply]
I created mini repo 
https://github.com/arturabg/pixi-loading

**Reply by SuperSodaSea** [reply]
Ummm, I tried and it seems that there is no error?
```
> node -v
v18.13.0
> npm -v 
9.4.1
```

**Reply by arturabg** [reply]
Yeah there is no error but editor show -> Unresolved variable eventMode , Unresolved function or method on() 

I do not had like type error before, when I updated the version I notice like error Unresolved function or method on()

<img width="645" alt="Screen Shot 2023-03-24 at 15 26 37" src="https://user-images.githubusercontent.com/83285441/227509567-2041d01a-4a20-40a4-9695-c06dcd42252a.png">

**Reply by bigtimebuddy** [reply]
VSCode worked fine for me. 

```
% npm -v
8.13.2
% node -v
v18.6.0
```

<img width="640" alt="Screen Shot 2023-03-24 at 8 52 59 PM" src="https://user-images.githubusercontent.com/864393/227669736-d94fea20-c044-4466-bd21-6e5a7adaef5a.png">

**Reply by arturabg** [reply]
Thank you very much for advice 💪✅

**Reply by arturabg** [reply]
<img width="721" alt="Screen Shot 2023-03-26 at 20 13 52" src="https://user-images.githubusercontent.com/83285441/227789095-b68dcff8-f8ef-4cec-93cd-68e1ff62ff7c.png">

I change this and it's work fine

**Reply by SuperSodaSea** [reply]
Maybe WebStorm(?) have poor integration with TypeScript 5.0? :thinking: Maybe you can try `"typescript": "^4.9.5"` and see if it works.

BTW you can try to add this fix mentioned in [Release note of PixiJS 7.2.1](https://github.com/pixijs/pixijs/releases/tag/v7.2.1), though I think it have nothing to do with your issue.

> **Warning**
> **About TypeScript 5.0**: There's a fix in this release to support TypeScript 5.0, however, PixiJS includes a dependency for `@types/css-font-loading-module` (0.0.7) this package that is only compatible with TS 4.x. For errors related to this dependency, you can suppress it by [overriding](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) the version to be `^0.0.8` in **package.json**:
> ```json
> {
>     "overrides": {
>         "@types/css-font-loading-module": "^0.0.8"
>     }
> }
> ```

**Reply by arturabg** [reply]
Pretty good. It's worked fine with "typescript": "^4.9.5". You were right WebStorm have poor integration. Lot of thanks for helping me.

<img width="702" alt="Screen Shot 2023-03-26 at 23 40 58" src="https://user-images.githubusercontent.com/83285441/227800742-a51e40ab-cf87-4f24-b2d5-4384af648002.png">

<img width="595" alt="Screen Shot 2023-03-26 at 23 44 43" src="https://user-images.githubusercontent.com/83285441/227800761-df73c479-4d67-40c1-a4ab-b1d37935d608.png">
